velox/exec/RowsStreamingWindowBuild.cpp

```diff
@@ -43,24 +43,45 @@ RowsStreamingWindowBuild::RowsStreamingWindowBuild(
       this);
 }
 
+bool RowsStreamingWindowBuild::needsInput() {
+  // No partitions are available or there is no rows in currentPartition.
+  return windowPartitions_.empty() || outputPartition()->numRows() == 0;
+}
+
+std::shared_ptr<WindowPartition> RowsStreamingWindowBuild::inputPartition()
+    const {
+  VELOX_CHECK(!windowPartitions_.empty());
+  VELOX_CHECK(!windowPartitions_.back()->complete());
```

Comment:
If we happen to call inputPartition() after the current input is marked complete,  it should still return correctly. What we want to avoid is to addInput to a complete partition, which ensureInputPartition() should take care of. So we can remove 
VELOX_CHECK(!windowPartitions_.back()->complete()); I think.

Reply:
It is better to have that check as inputPartition() is invoked to add input.